web/src/lib/components/asset-viewer/video-viewer/video-native-viewer.svelte

```diff
@@ -0,0 +1,150 @@
+<script lang="ts">
+  import FaceEditor from '$lib/components/asset-viewer/face-editor/face-editor.svelte';
+  import VideoLayout from '$lib/components/asset-viewer/video-viewer/video-layout.svelte';
+  import { isFaceEditMode } from '$lib/stores/face-edit.svelte';
+  import { loopVideo as loopVideoPreference, videoViewerMuted, videoViewerVolume } from '$lib/stores/preferences.store';
+  import { getAssetPlaybackUrl, getAssetThumbnailUrl } from '$lib/utils';
+  import { handleError } from '$lib/utils/handle-error';
+  import { AssetMediaSize } from '@immich/sdk';
+  import { tick } from 'svelte';
+  import type { SwipeCustomEvent } from 'svelte-gestures';
+  import { swipe } from 'svelte-gestures';
+  import { t } from 'svelte-i18n';
+  import { fade } from 'svelte/transition';
+  import type { MediaAutoPlayFailEvent, MediaVolumeChangeEvent } from 'vidstack';
+  import 'vidstack/player';
+  import 'vidstack/player/styles/base.css';
+  import 'vidstack/player/ui';
+
+  import type { MediaPlayerElement } from 'vidstack/elements';
+  interface Props {
+    assetId: string;
+    loopVideo: boolean;
+    cacheKey: string | null;
+    onPreviousAsset?: () => void;
+    onNextAsset?: () => void;
+    onVideoEnded?: () => void;
+    onVideoStarted?: () => void;
+    onControlsChange?: ({ controlsVisible }: { controlsVisible: boolean }) => void;
+  }
+
+  let {
+    assetId,
+    loopVideo,
+    cacheKey,
+    onPreviousAsset = () => {},
+    onNextAsset = () => {},
+    onVideoEnded = () => {},
+    onVideoStarted = () => {},
+    onControlsChange = () => {},
+  }: Props = $props();
+
+  let player: MediaPlayerElement | undefined = $state();
+
+  let assetFileUrl = $derived(getAssetPlaybackUrl({ id: assetId, cacheKey }));
+  let videoElement = $derived(player?.querySelector('video') as HTMLVideoElement);
+
+  let forceMuted = $state(false);
+  let containerWidth = $state(0);
+  let containerHeight = $state(0);
+
+  const streamType = 'on-demand';
+  const logLevel: 'silent' | 'error' | 'warn' | 'info' | 'debug' = 'error';
+
+  const onSwipe = (event: SwipeCustomEvent) => {
+    if (event.detail.direction === 'left') {
+      onNextAsset();
+    }
+    if (event.detail.direction === 'right') {
+      onPreviousAsset();
+    }
```

Comment:
Could argue that an else if (or probably even nicer a switch) would look better here because it'd emphasize that those are always mutually exclusive. Minor thing though given this part is trivial